lib/kafo/data_types/boolean.rb

```diff
@@ -0,0 +1,27 @@
+module Kafo
+  module DataTypes
+    class Boolean < DataType
+      def typecast(value)
+        case value
+          when '0', 'false', 'f', 'n', false
+            false
+          when '1', 'true', 't', 'y', true
+            true
+          else
+            value
+        end
+      end
+
+      def valid?(input, errors = [])
+        if input.is_a?(::TrueClass) || input.is_a?(FalseClass)
```

Comment:
tap's available since 1.8.7 so I've rewritten it that way, and added the missing namespace `::`.  (tap wasn't in 1.8.5 on EL5)